FIX: Legacy expression tool (#1405)
telefonicaid/iotagent-node-lib: scripts/legacy_expression_tool/README.md

| `--mongouri`           | The MongoDB URI to connect to                                                                            | `mongodb://localhost:27017/` | No        |
| `--database`           | The database name to replace the expressions                                                             | NA                           | Yes       |
| `--collection`         | The collection name to replace the expressions                                                           | NA                           | Yes       |
| `--translation`        | The translation dictionary file to replace the expressions                                               | `translation.json`           | No        |
| `--debug`              | Enable debug mode                                                                                        | `False`                      | No        |
| `--commit`             | Commit the changes to the database                                                                       | `False`                      | No        |

> | `--expressionlanguage` | What to do with the expression language field. Possibles values: `delete`, `ignore`, `jexl` or `jexlall` | `ignore`                     | No        |

Reviewer: ```suggestion
| `--expressionlanguage` | What to do with the expression language field. Possibles values: `delete` (to delete the field), `ignore` (to leave the field untouched), `jexl` (...) or `jexlall` (...) | `ignore`                     | No        |
```

The `(...)` has to be filled. Not sure the difference between `jexl` and `jexlall` so I cannot complete myself...

Author: ```suggestion
| `--expressionlanguage` | What to do with the expression language field. Possibles values: `delete` (to delete the field), `ignore` (to leave the field untouched), `jexl` (to force to jexl on all groups that match the conditions) or `jexlall` (that, additionally, includes all groups that have a different expression language value than jexl to the query) | `ignore`                     | No        |
```

Reviewer: ```suggestion
| `--expressionlanguage` | What to do with the expression language field. Possibles values: `delete` (to delete the field), `ignore` (to leave the field untouched), `jexl` (to force to jexl on all groups/devices that match the conditions) or `jexlall` (like `jext` but additionally including all groups/devices that have a different expression language value than jexl to the query) | `ignore`                     | No        |
```

> | `--expressionlanguage` | What to do with the expression language field. Possibles values: `delete` (to delete the field), `ignore` (to leave the field untouched), `jexl` (to force to jexl on all groups/devices that match the conditions) or `jexlall` (like `jexl` but additionally including all groups/devices that have `expressionLanguage` field to the query) | `ignore`                     | No        |

Reviewer: ```suggestion
| `--expressionlanguage` | What to do with the expression language field. Possibles values: `delete` (to delete the `expressionLanguage` field), `ignore` (to leave the field untouched, **FIXME** don't use, it can leave documents inconsistent), `jexl` (to change expressions to JEXL and in addition to change `expressionLanguage` field to `jexl` when the group/device uses legacy in some of the expression-related field, no matter if the group/devices have previously the `expresionLanguage` field or not) or `jexlall` (like `jexl` option, but also including groups/devices that doesn't use JEXL i any expression-related field but have `expresionLanguage` field defined, no matter its value) | `ignore`                     | No        |
```

    --translation <translation-file> \
    --commit
```

### Output statistics

When the script is executed, it prints some statistics about the matches found. This statistics can be printed filtered
by service or subservice. By default, no statistics are printed. You can change it using the `--statistics` plus the
aggregation mode. The possible values are `service` and `subservice`.

```bash
Found 2 legacy expressions in 2 documents
                                                        _id
service                                         service1 service2 All
expression
${@value*100/total}                                    1        1   2
All                                                    1        1   2
```

### Known issues

#### Execution with `expressionlanguage` set to `jexlall`

When executing the script with `expressionlanguage` set to `jexlall`, the script will look for all the documents

> containing legacy expressions or the parameter `expressionLanguage`. This would change the number of documents found,

Reviewer: What "containing legacy expression" mean? Should be detailed.

I understand that when `jexl` is used the statistics are accurate. If I'm right, then I'd suggest to explicitly say it here.

Author: Solved in https://github.com/telefonicaid/iotagent-node-lib/pull/1405/commits/acc754a5baf9076af420431aade4b6cd76e46ea1

> and the statistics will be different.

Author: ```suggestion
and the statistics will be different. Running the script with the option set to `jexl` would not have impact on  the statistics. 
```

#### Replacing expression without setting jexl at group or device level

> When executing the script without setting `expressionlanguage` to `jexl` or `jexlall` (or `delete`), the script will not
> change the`expressionLanguage` field in the document. This means that the legacy expressions will be replaced, but the

Reviewer: ```suggestion
When executing the script setting `--expressionlanguage` to `ignore` (or when `--expressionlanguage` is not used), the script will not
change the`expressionLanguage` field in the document. This means that the legacy expressions will be replaced, but the
```

I think is more clear that way.
